src/test/java/org/elasticsearch/search/aggregations/reducers/DerivativeTests.java

```diff
@@ -509,111 +295,32 @@ public void singleValuedFieldWithGaps_insertZeros() throws Exception {
-        assertThat(bucket.getDocCount(), equalTo(0l));
-        docCountDeriv = bucket.getAggregations().get("deriv");
-        assertThat(docCountDeriv, notNullValue());
-        assertThat(docCountDeriv.value(), equalTo(-2d));
-
-        bucket = buckets.get(7);
-        assertThat(bucket, notNullValue());
-        assertThat(((Number) bucket.getKey()).longValue(), equalTo(7l));
-        assertThat(bucket.getDocCount(), equalTo(0l));
-        docCountDeriv = bucket.getAggregations().get("deriv");
-        assertThat(docCountDeriv, notNullValue());
-        assertThat(docCountDeriv.value(), equalTo(0d));
-
-        bucket = buckets.get(8);
-        assertThat(bucket, notNullValue());
-        assertThat(((Number) bucket.getKey()).longValue(), equalTo(8l));
-        assertThat(bucket.getDocCount(), equalTo(0l));
-        docCountDeriv = bucket.getAggregations().get("deriv");
-        assertThat(docCountDeriv, notNullValue());
-        assertThat(docCountDeriv.value(), equalTo(0d));
-
-        bucket = buckets.get(9);
-        assertThat(bucket, notNullValue());
-        assertThat(((Number) bucket.getKey()).longValue(), equalTo(9l));
-        assertThat(bucket.getDocCount(), equalTo(3l));
-        docCountDeriv = bucket.getAggregations().get("deriv");
-        assertThat(docCountDeriv, notNullValue());
-        assertThat(docCountDeriv.value(), equalTo(3d));
-
-        bucket = buckets.get(10);
-        assertThat(bucket, notNullValue());
-        assertThat(((Number) bucket.getKey()).longValue(), equalTo(10l));
-        assertThat(bucket.getDocCount(), equalTo(2l));
-        docCountDeriv = bucket.getAggregations().get("deriv");
-        assertThat(docCountDeriv, notNullValue());
-        assertThat(docCountDeriv.value(), equalTo(-1d));
-
-        bucket = buckets.get(11);
-        assertThat(bucket, notNullValue());
-        assertThat(((Number) bucket.getKey()).longValue(), equalTo(11l));
-        assertThat(bucket.getDocCount(), equalTo(1l));
-        docCountDeriv = bucket.getAggregations().get("deriv");
-        assertThat(docCountDeriv, notNullValue());
-        assertThat(docCountDeriv.value(), equalTo(-1d));
+        List<Bucket> buckets = deriv.getBuckets();
+        assertThat(buckets.size(), equalTo(valueCounts_empty.length));
+
+        for (int i = 0; i < valueCounts_empty.length; i++) {
+            Histogram.Bucket bucket = buckets.get(i);
+            checkBucketValues("Bucket " + i + ": ", bucket, i, valueCounts_empty[i]);
+            SimpleValue docCountDeriv = bucket.getAggregations().get("deriv");
+            if (firstDerivValueCounts_empty[i] == null) {
+                assertThat(docCountDeriv, nullValue());
+            } else {
+                assertThat(docCountDeriv.value(), equalTo(firstDerivValueCounts_empty[i]));
+            }
+        }
     }
 
+    private void checkBucketValues(final String msg, final Histogram.Bucket bucket, final long key, final long docCount) {
```

Comment:
Could we call the parameters here `expectedKey` and `expectedDocCount` to be clearer on what they are?